src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,77 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+
     public void getStatistic(String fromFileName, String toFileName) {
+        if (fromFileName == null) {
+            System.out.println("Source file doesn't exists");
+            return;
+        }
+        if (toFileName == null) {
+            System.out.println("Destination file doesn't exists");
+            return;
+        }
+
+        File file = new File(fromFileName);
+        StringBuilder builder = new StringBuilder();
+        int sumSupply = 0;
+        int sumBuy = 0;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            String value = reader.readLine();
+            while (value != null) {
+                builder.append(value).append(System.lineSeparator());
+                value = reader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can`t read file", e);
+        }
+
+        String[] splitValues = builder.toString().split(System.lineSeparator());
+
+        for (String value : splitValues) {
+            if (value == null) {
+                System.out.println("From file name is empty!");
+            }
+
+            assert value != null;
```

Comment:
Using `assert` for checking `null` values is not recommended in production code as assertions can be disabled. Consider using a proper `if` statement to handle unexpected `null` values.